lib/traject/macros/dor.rb

```diff
@@ -16,11 +16,18 @@ def resource_images_iiif_urls
         content_metadata = resource.public_xml.at_xpath('/publicObject/contentMetadata')
         next if content_metadata.nil?
 
-        images = content_metadata.xpath('resource/file[@mimetype="image/jp2"]')
-        jp2s = images.select { |node| node.attr('id') =~ /jp2$/ }
+        # Select conventional file images or virtual external ones
+        images = content_metadata.xpath(
+          '(resource/file[@mimetype="image/jp2"] | resource/externalFile[@mimetype="image/jp2"])'
+        )
+        # Allow for selection of conventional ids and fileId for virtual objects
+        jp2s = images.select { |node| (node.attr('id') || node.attr('fileId')) =~ /jp2$/ }
```

Comment:
Would it make sense to do a `collect` here so that `jp2s` is already the id that we want (and saves us from having to duplicate the logic below).  Just a thought.

Reply:
Ah, I see, we're also getting the `objectId` attribute below...  This feels a bit nasty, but don't have a succinct solution off the top of my head to suggest.